src/components/modal/ModalCreateBooks.tsx

```diff
@@ -0,0 +1,188 @@
+        reader.readAsDataURL(files[0])
+        setFileImg(files[0])
+        reader.onloadend = function () {
+          const base64data = reader.result
+          if (base64data) setCoverImg(base64data as string)
+        }
+      }
+    }
+  }
+  const hanlderCreateBooks = async (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault()
+    try {
+      const form = new FormData()
+      form.append('Title', booksData.Title)
+      form.append('Description', booksData.Description)
+      form.append('Isbn', booksData.Isbn)
+      form.append('Quantity', booksData.Quantity.toString())
+      form.append('Cover', fileImg)
+      form.append('PublishDate', booksData.PublishDate)
+      booksData.AuthorIds.forEach((author) => form.append('AuthorIds', author.Id.toString()))
+      await BookService.createBook(form)
+    } catch(error) {
+      if(axios.isAxiosError(error)) {
+        if(error.response?.status === 401) {
+          console.log('Not authenticated')
+        }
+      }
+    }
+  }
+
+  const handleCreateAuthor = async (event: FormEvent<HTMLFormElement>) => {
+    handlerNewAuthor()
+    event.preventDefault()
+    try {
+      const form = new FormData()
+      form.append('FirstName', authorData.FirstName)
+      form.append('LastName', authorData.LastName)
+      await AuthorService.createAuthor(form)
+    } catch(error) {
+      if(axios.isAxiosError(error)) {
+        if(error.response?.status === 401) {
+          console.log('Not authenticated')
+        }
+      }
+    }
+  }
+
+  const handlerNewAuthor = () => {
+    AuthorService.getAuthors().then((response) => {
+      setAtuhors(response)
+    })
+    setShowAddAuthor(true)
+    setShowAddButton(true)
+  }
+
+  const handlerCloseModal = () => {
+    onClose()
+  }
+
+  const handleFormAuthor = () => {
+    setShowAddAuthor(false)
+    setShowAddButton(false)
+  }
+
+  const handleCloseFormAuthor = () => {
+    setShowAddAuthor(true)
+    setShowAddButton(true)
+  }
+
+  useEffect(() => {
+    AuthorService.getAuthors()
+      .then(response => {
+        setAtuhors(response)
+      })
+  }, [])
+
+  const handleAuthor = (newAuthors: MultiValue<Author>) => {
+    setBooksData((prev) => ({ ...prev, AuthorIds: newAuthors as Author[] }))
+  }
+
+  if(!open) return null
+
+
+  return ReactDOM.createPortal(
+    <div className='overlay'>
+      <div className='content'>
+        <form className='form_for_modal' onSubmit={hanlderCreateBooks}>
+          <div className='heder_modal'>
+            <button type='button' onClick={handlerCloseModal} className='x_button_for_modal'><XIconForModal className='x_icon_for_modal'/></button>
+          </div>
+          <img className='modal_cover_img' src={coverImg ? coverImg : placeholder} />
+          <div className='container_for_element'>
+            <input type='text' className='input_for_title' placeholder='Title' onChange={({ target }) => setBooksData((prevTitle) => ({ ...prevTitle, Title: target.value }))}/>
```

Comment:
Break into multiple lines if you have multiple properties - same goes for the whole file

Reply:
Done.